packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -47,8 +47,11 @@ export interface TerminalWidgetFactoryOptions extends Partial<TerminalWidgetOpti
 export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget {
 
     private readonly TERMINAL = 'Terminal';
+    isInteractiveTerminal = true;
+    termId = -1;
     protected readonly onTermDidClose = new Emitter<TerminalWidget>();
-    protected terminalId = -1;
+    protected readonly onTermDidDisposeEmitter = new Emitter<void>();
```

Comment:
I would prefer if we add `onDidDisposeEmitter` to `BaseWidget` instead that it is available for each widget. Instead of inventing it each time.